go/vt/vtgate/planbuilder/testdata/dml_cases.txt

```diff
@@ -3187,3 +3299,405 @@ Gen4 plan same as above
+    "MultiShardAutocommit": false,
+    "TableName": "user_extra",
+    "VindexOffsetFromSelect": {
+      "user_index": "[0]"
+    },
+    "Inputs": [
+      {
+        "OperatorType": "Route",
+        "Variant": "Scatter",
+        "Keyspace": {
+          "Name": "user",
+          "Sharded": true
+        },
+        "FieldQuery": "select id from `user` where 1 != 1",
+        "Query": "select id from `user` for update",
+        "Table": "`user`"
+      }
+    ]
+  }
+}
+Gen4 plan same as above
+
+# insert using select with auto-inc column using vitess sequence, sequence column present
+"insert into user_extra(id, user_id) select null, id from user"
+{
+  "QueryType": "INSERT",
+  "Original": "insert into user_extra(id, user_id) select null, id from user",
+  "Instructions": {
+    "OperatorType": "Insert",
+    "Variant": "Select",
+    "Keyspace": {
+      "Name": "user",
+      "Sharded": true
+    },
+    "TargetTabletType": "PRIMARY",
+    "AutoIncrement": "main:2",
+    "MultiShardAutocommit": false,
+    "TableName": "user_extra",
+    "VindexOffsetFromSelect": {
+      "user_index": "[1]"
+    },
+    "Inputs": [
+      {
+        "OperatorType": "Route",
+        "Variant": "Scatter",
+        "Keyspace": {
+          "Name": "user",
+          "Sharded": true
+        },
+        "FieldQuery": "select null, id from `user` where 1 != 1",
+        "Query": "select null, id from `user` for update",
+        "Table": "`user`"
+      }
+    ]
+  }
+}
+Gen4 plan same as above
+
+# sharded insert from select
+"insert into user(id) select 1 from dual"
+{
+  "QueryType": "INSERT",
+  "Original": "insert into user(id) select 1 from dual",
+  "Instructions": {
+    "OperatorType": "Insert",
+    "Variant": "Select",
+    "Keyspace": {
+      "Name": "user",
+      "Sharded": true
+    },
+    "TargetTabletType": "PRIMARY",
+    "AutoIncrement": "main:0",
+    "MultiShardAutocommit": false,
+    "TableName": "user",
+    "VindexOffsetFromSelect": {
+      "costly_map": "[-1]",
+      "name_user_map": "[-1]",
+      "user_index": "[0]"
+    },
+    "Inputs": [
+      {
+        "OperatorType": "Projection",
+        "Columns": [
+          "1"
+        ],
+        "Expressions": [
+          "INT64(1)"
+        ],
+        "Inputs": [
+          {
+            "OperatorType": "SingleRow"
+          }
+        ]
+      }
+    ]
+  }
+}
+Gen4 plan same as above
+
+# insert using select with sharding column is autoinc and not present in the insert column query
+"insert into user(pattern) SELECT 1"
+{
+  "QueryType": "INSERT",
+  "Original": "insert into user(pattern) SELECT 1",
+  "Instructions": {
+    "OperatorType": "Insert",
+    "Variant": "Select",
+    "Keyspace": {
+      "Name": "user",
+      "Sharded": true
+    },
+    "TargetTabletType": "PRIMARY",
+    "AutoIncrement": "main:1",
+    "MultiShardAutocommit": false,
+    "TableName": "user",
+    "VindexOffsetFromSelect": {
+      "costly_map": "[-1]",
+      "name_user_map": "[-1]",
```

Comment:
I think we can exclude those `VindexOffsetFromSelect` that are unknown (with the value `-1`) from the plan output